### ISSUE TITLE
Exception when trying to edit a variant without a default price

### ISSUE THREAD

**Issue by oneiros** [solidusio/solidus#2924]
### Steps to reproduce

Configure a default currency in `config/initializers/spree.rb`:

```ruby
  config.currency = "EUR"
```

Log into the admin backend, select a product and on the tab "Variants" add a new variant.

Go to the tab "Prices" and under "Variant Pricing" edit the price of the newly created variant.

Set the Currency to "USD" and save the price.

Go back to the "Variants" tab and edit the variant.

### Expected behavior

At the very least, no exception should be raised

### Actual behavior

The following exception is raised:

```
ActionView::Template::Error (undefined method `currency' for nil:NilClass):
    65:       <div class="col-3">
    66:         <div class="field" data-hook="price">
    67:           <%= f.label :price %>
    68:           <%= render "spree/admin/shared/number_with_currency", f: f, amount_attr: :price, currency: @variant.default_price.currency %>
    69:         </div>
    70:       </div>
    71: 
  
solidus_backend (2.7.0) app/views/spree/admin/variants/_form.html.erb:68:in `_779db0d19337b31fccae3d69b3329154'
```

The reason is quite obvious: The partial expects a `default_price` to always be present. The `default_price` is being determined in ` solidus/core/app/models/concerns/spree/default_price.rb` using, amongst others, the scope `with_default_attributes`. This in turn uses the configured currency. So if no price exists for the configured currency, no `default_price` can be found.

I guess a possible solution would be to fix the view partial to not raise an exception (and I would be happy to prepare a PR for that), but I wonder if it should even be possible for a variant to not have a `default_price`. If no, the actual fix would be quite different.

### System configuration

**Solidus Version**: 2.7.0

**Extensions in use**:

* 'solidus_product_assembly'
* 'solidus_print_invoice'
* 'solidus_reviews'

**Comment by jacobherrington** [comment]
I think not raising an exception is a good idea. I'm not too familiar with doing international stores, so anyone can correct me, but I think people should be able to edit prices for different currencies in this way.

**Comment by jtapia** [comment]
I created this PR https://github.com/solidusio/solidus/pull/3067 that fixed that issue, let me know your thoughts

**Comment by kennyadsl** [comment]
Closed with #2504, thanks for reporting the issue.
